### PR TITLE
Replace issue templates with issue forms [SDK-4157]

### PR DESCRIPTION
**Pull request by Widcket**
### Changes

This PR replaces the Markdown-based issue templates with [issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms). These forms are a direct conversion of the Markdown issue templates, with the addition of a checklist at the top containing links to self-serve resources.

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [X] All existing and new tests complete without errors

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 blank_issues_enabled: false
 contact_links:
   - name: Auth0 Community
-    url: https://community.auth0.com/c/sdks/5
```
Comment (Widcket) [inline .github/ISSUE_TEMPLATE/config.yml:4]: The path `/c/sdks/5` no longer exists.

File: .github/ISSUE_TEMPLATE/Bug Report.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        - Windows
+    validations:
+      required: true
+
+  - type: input
+    id: environment-platform-version
+    attributes:
+      label: Platform version(s)
+    validations:
+      required: true
```
Comment (Widcket) [inline .github/ISSUE_TEMPLATE/Bug Report.yml:79]: Please check that these fields are appropriate and sufficient. Let me know if you'd prefer to change them or add more.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 1 above)
Comment (frederikprijck) [reply .github/ISSUE_TEMPLATE/config.yml:4]: Should we use https://community.auth0.com/tag/auth0-oidc-client-net instead?

Comment (Widcket) [reply .github/ISSUE_TEMPLATE/config.yml:4]: Given that the forums include a [FAQ](https://community.auth0.com/c/faq/42) section, it'd be wise to encourage developers do a site-wide search, as their problem might not be specific to this SDK, but could be answered in one of those FAQs.

**Pull request merged**